python/ray/serve/async_goal_manager.py

```diff
@@ -11,9 +11,6 @@ class AsyncGoalManager:
     def __init__(self):
         self._pending_goals: Dict[GoalId, asyncio.Event] = dict()
 
-    def get_pending_goal_ids(self) -> List[GoalId]:
```

Comment:
some random cleanup; this was only used for checkpointing and was redundant with another data structure already being checkpointed